code/config.py

```diff
@@ -1,120 +1,360 @@
-save_interval = 1
-checkpoints_folder = 'checkpoints'		 			# where models are saved
-pretrain = False 									# imagenet pretrain?
-log_folder = 'logs'
-log_csv = get_log_csv_name(log_folder) 				# is named with date and time
+#######################################################
+#               ARGUMENTS FROM ARGPARSE               #
+#######################################################
+args = parser.parse_args()
 
-###########################################
-#               PREDICTION                #
-###########################################
+# Device to use for PyTorch code.
+device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
 
-# selecting the best model
-auto_select = True 						# automatically selects the model with the highest validation accuracy
-eval_model = 'checkpoints/xyz.pt' 		# does nothing if auto_select is true
+# Automatically read in the classes.
+classes = get_classes(folder=args.all_wsi)
+num_classes = len(classes)
 
-preds_train = 'preds_train' 			# where to put the training prediction csv files
-preds_val = 'preds_val' 				# where to put the validation prediction csv files
-preds_test = 'preds_test' 				# where to put the testing prediction csv files
+# This is the input for model training, automatically built.
+train_patches = args.train_folder.joinpath("train")
+val_patches = args.train_folder.joinpath("val")
 
-###########################################
-#               EVALUATION                #
-###########################################
+# Compute the mean and standard deviation for the given set of WSI for normalization.
+path_mean, path_std = compute_stats(folderpath=args.all_wsi)
 
-# find the best threshold for filtering noise (discard patches with a confidence less than this threshold)
-threshold_search = [0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9]
+# Only used is resume_checkpoint is True.
+resume_checkpoint_path = args.checkpoints_folder.joinpath(args.checkpoint_file)
+
+# Named with date and time.
+log_csv = get_log_csv_name(log_folder=args.log_folder)
 
-# folder for outputing wsi predictions based on each threshold
-inference_train = 'inference_train'
-inference_val = 'inference_val'
-inference_test = 'inference_test'
+# Does nothing if auto_select is True.
+eval_model = args.checkpoints_folder.joinpath(args.checkpoint_file)
+
+# Find the best threshold for filtering noise (discard patches with a confidence less than this threshold).
+threshold_search = [0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9]
 
-# for visualization
-colors = ['red', 'white', 'blue', 'green', 'purple', 'orange', 'black', 'pink', 'yellow'] 	# this order is the same order as your sorted classes
-vis_train = 'vis_train'
-vis_val = 'vis_val'
-vis_test = 'vis_test'
+# For visualization.
+# This order is the same order as your sorted classes.
+colors = [
+    "red", "white", "blue", "green", "purple", "orange", "black", "pink",
+    "yellow"
```

Comment:
why is this on a separate line

Reply:
@jasonwei20 @ntomita I will change this to a tuple since it should be immutable.

Reply:
Changed to tuple in newest commit.